akka-remote/src/main/scala/akka/remote/RemoteActorRefProvider.scala

```diff
@@ -620,6 +621,24 @@ private[akka] class RemoteActorRefProvider(
         local.addressString
     }
   }
+
+  override lazy val systemUid: Long = {
+    val address = getDefaultAddress
+
+    val sb = new StringBuilder
+    sb.append(address.host.getOrElse(""))
+      .append(address.port.getOrElse(0))
+      .append(System.currentTimeMillis())
+      .append(System.nanoTime())
+    val key = sb.toString()
+
+    val hash1 = MurmurHash3.stringHash(key.substring(0, key.length / 2))
+    val hash2 = MurmurHash3.stringHash(key.substring(key.length / 2))
+    val longHash = hash1.toLong << 32 | hash2
+
+    longHash
```

Comment:
@johanandren @pvlugter Hashing instead

Reply:
Is hashing half the string to int twice and using as parts of long really the same as having a long hash to start with?

Reply:
I think that sounds good, an internal copy/impl.

Reply:
added port of 128 bit hashing

Reply:
If we did want to avoid adding another hashing function, I think there could be effective ways to combine 32-bit hashes. We expect the timestamps to be different, but know that if the timestamp is exactly the same, then either the host or the port are different. So could make combinations for each half that should always be unique. Could also get new timestamps for the second half, which may have shifted by then in an unpredictable way. So `long(hash32(host + time1), hash32(port + time2))` seems that it would create reasonable uniqueness. And guaranteed to have unique combinations, even if the times are identical across nodes.

But having the 128-bit (and 64-bit) hashing around seems useful anyway.

Reply:
Looking closer at the math I see that it's the same probability of getting two hash collisions from two 32 bit as from one 64 bit.
* https://stackoverflow.com/a/62667633
* https://en.wikipedia.org/wiki/Birthday_problem#Probability_table

Given "perfect" hashing, 50% risk of a collision is 1 in 2^(64/2) = 4294967296
Which is the same as 2^(32/2) * 2^(32/2)

Then I agree, that there is not much point in using 64 bit hashing, and slitting the parts as you say sounds good. We could even create the first timestamp when the RARP is created, and the second when systemUid is called. Then there is plenty of cpu scheduling and other execution randomness (e.g. binding port) between the two timestamps.